### DISCUSSION TITLE
gif animation export "precise frame control"

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#349] [Ideas]
The `py5_tool.animated_gif` feature is great, but I miss from the _gifAnimation_ library that I used with Processing Java/Python mode the "fine control" of choosing to add frames in some circumstances...

```python
from py5_tools import animated_gif

n = 10

def setup():
    size(500, 500)
    no_stroke()
    frame_rate(16)
    # filename, number of frames, sample interval, frame duration in gif
    animated_gif('test.gif', n, 0.05, 1)
        
def draw():
    background(100)
    fill(255)
    text_size(40)
    text(frame_count, 200, 200)
    
    
    if frame_count > n + 5:
        exit_sketch()
```
![test](https://github.com/py5coding/py5generator/assets/3694604/92432361-b851-45fc-a2f5-8f5e68e03fc6)

When making small loops the "time control" afforded by `animated_gif` doesn't seem helpful. I then usually export all frames (or every odd frame) and generate the animation elsewhere. Am I missing somethig? 

A lazy idea... could we have `animated_gif_from_frames(filename, list_of_frames, duration)`?

**Comment by hx2A** [comment]
What if instead of a time control you listed the frame numbers you wanted to have included? Like if you wanted the first 20 odd numbered frames, you could pass a kwarg `frame_nums=range(1, 40, 2)` or something like that? Then just the frames 1, 3, 5, ... 39 etc are included?

**Reply by villares** [reply]
Exactly! That's what I meant with "list_of_frames". That would be great!

**Reply by hx2A** [reply]
OK! This should be very achievable. I will open an issue to track this.
